Log non-fatal accessing repos errors instead of failing

cam72cam opened the pull request:
Longer term we need to track these and likely remove them after a grace period

Fixes https://github.com/opentofu/registry/issues/66

I also reverted some previous changes to remove providers/modules that were having either temporary or permanent issues.

Yantrio commented:
Hi @alrs, thanks for this contribution! I'll merge this once I have the workflow for tests passing 👍

cam72cam committed:
Revert "Delete modules that no longer have active repositories"

This reverts commit df6ff08caac4cc4329a0248c5672f913dda14d17.

Signed-off-by: Christian Mesh <christianmesh1@gmail.com>

cam72cam committed:
Revert "Remove missing modules"

This reverts commit 349e323a703640261abbb68ea8ec5a6d1277ace8.

Signed-off-by: Christian Mesh <christianmesh1@gmail.com>

cam72cam committed:
Revert "Removed modules that have no tags and/or are no longer on github"

This reverts commit b862f54fd186ff1cb72b1a0d59a481cd421a481b.

Signed-off-by: Christian Mesh <christianmesh1@gmail.com>

cam72cam committed:
Log non-fatal accessing repos errors instead of failing

Longer term we need to track these and likely remove them after a grace period

Signed-off-by: Christian Mesh <christianmesh1@gmail.com>

cube2222 commented on src/internal/module/update.go:
```diff
@@ -53,11 +57,15 @@ func (p Module) getSemVerTagsFromRSS() ([]string, error) {
-		return "", err
+		p.Logger.Error("Unable to fetch tags, skipping", slog.Any("err", err))
+		return "", nil
+		//return "", err
```
leftover comment

cam72cam replied:
I did this intentionally as I see it as a stopgap

cube2222 replied:
Could you add a TODO comment then?

cam72cam replied:
Yep

cube2222 commented on src/internal/module/update.go:
```diff
@@ -53,11 +57,15 @@ func (p Module) getSemVerTagsFromRSS() ([]string, error) {
-		return "", fmt.Errorf("no semver tags found in repository %s", p.RepositoryURL())
+		p.Logger.Error("no semver tags found in repository, skipping", slog.String("url", p.RepositoryURL()))
+		return "", nil
+		//return "", fmt.Errorf("no semver tags found in repository %s", p.RepositoryURL())
```
leftover comment

cube2222 commented on src/internal/provider/update.go:
```diff
@@ -77,11 +82,15 @@ func (p Provider) getSemVerTagsFromRSS() ([]string, error) {
-		return "", err
+		p.Logger.Error("Unable to fetch tags, skipping", slog.Any("err", err))
+		return "", nil
+		//return "", err
```
leftover comment

cube2222 commented on src/internal/provider/update.go:
```diff
@@ -77,11 +82,15 @@ func (p Provider) getSemVerTagsFromRSS() ([]string, error) {
-		return "", fmt.Errorf("no semver tags found in repository %s", p.RepositoryURL())
+		p.Logger.Error("no semver tags found in repository, skipping", slog.String("url", p.RepositoryURL()))
+		return "", nil
+		//return "", fmt.Errorf("no semver tags found in repository %s", p.RepositoryURL())
```
leftover comment

cam72cam commented on src/internal/provider/update.go:
```diff
@@ -77,11 +82,15 @@ func (p Provider) getSemVerTagsFromRSS() ([]string, error) {
 	if err != nil {
-		return "", err
+		p.Logger.Error("Unable to fetch tags, skipping", slog.Any("err", err))
+		return "", nil
```
I was keeping with the current "pattern" in the codebase that doesn't use custom error types.  I'm hoping to rewrite a lot of this code and was trying to keep the changeset minimal.

cam72cam replied:
Not saying it's a good pattern...

cam72cam committed:
Commented out code -> TODOs

Signed-off-by: Christian Mesh <christianmesh1@gmail.com>
